src/goto-harness/memory_snapshot_harness_generator.cpp

```diff
@@ -210,10 +227,16 @@ code_blockt memory_snapshot_harness_generatort::add_assignments_to_globals(
   code_blockt code;
   for(const auto &pair : snapshot)
   {
-    const symbolt &symbol = pair.second;
-    if(!symbol.is_static_lifetime)
+    const symbolt &snapshot_symbol = pair.second;
+    if(!snapshot_symbol.is_static_lifetime)
       continue;
 
+    symbol_tablet &symbol_table = goto_model.symbol_table;
+    const symbolt &symbol =
+      (symbol_table.lookup(snapshot_symbol.base_name) != nullptr)
+        ? snapshot_symbol
+        : fresh_symbol_copy(snapshot_symbol, symbol_table);
```

Comment:
The `memory-analyzer` produces a symtab with `tmp` variables that the malloc(ed) pointers point to. These new variables do not exists in the "local" symbol table. So I add them here.

Reply:
The comment in the code is not so clear on this point, although it mentions it.

I would prefer some text similar to the above specifically mentioning this within the context of memory-analyzer.